### ISSUE TITLE
Is Pinecone even used in this.....? (confusing)

### ISSUE THREAD

**Issue by jackgofff** [AntonioErdeljac/next13-ai-companion#2]
I see `lib/memory.ts` implements the Pinecone Client and defines the `vectorSearch` function, but when running the app, my Pinecone index is displaying 0 vectors (see below)

From the Console logs it appears the chat history is just being handled by an array of strings, which is a huge problem for prompts > 2048 tokens. 

Great tutorial btw.

<img width="720" alt="image" src="https://github.com/AntonioErdeljac/next13-ai-companion/assets/141548271/d253f4d4-6abe-4c48-a769-02624db91e88">

**Comment by jackgofff** [comment]
Any update?

**Comment by rozer821** [comment]
same curiosity here

**Comment by Daniel-W1** [comment]
yes we are not inserting any data, but we are always searching in empty db ? To fix that i added an upsertfunction in the memory manager that converts the recent chat history to vector using openaiembedding and push it to pinecone. It seems to work fine. and the similarity searchs are returing something atleast ( not always empty )

**Comment by oxillix** [comment]
@Daniel-W1 Thank you for your comment! Would you mind sharing the code? I can't seem to get it working, it's my first LLM project :p

**Comment by Daniel-W1** [comment]
No probs, u can find the code here. https://github.com/Daniel-W1/ai-friends/blob/master/src/lib/memory.ts

This is where I used it, https://github.com/Daniel-W1/ai-friends/blob/master/src/app/api/chat/%5BchatId%5D/route.ts

I hope it helps, I am also no expert here :) any comment is appreciated!

**Comment by oxillix** [comment]
Thank you very much @Daniel-W1! Great to see how you used ChatGPT-3.5 instead of llama2-13b, i've been experimenting with llama2-70b, with mixed results. 

FYI:
This project is a fork of https://github.com/a16z-infra/companion-app
The UI/UX inspiration of this project comes from: https://beta.character.ai/

It's amazing to see how much knowledge is packed in this 6h tutorial!
